oem/supermicro/manager.go

```diff
@@ -0,0 +1,52 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package supermicro
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/stmcginnis/gofish/common"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+type Manager struct {
+	redfish.Manager
+	Oem ManagerOem
```

Comment:
It might be a simple interface to just get rid of this `Oem` field and move the fields of `ManagerOem` up a level into `Manager`.

Reply:
Yes, but it's an unnecessary detail for consumers. In the JSON unmarhalling it can pull out the fields and set them as direct properties of the manager struct.

Reply:
Yep. You're already using an OEM object, so it feels a little awkward to have to have OEMObject.OEM.Foo instead of OEMObject.Foo.

Reply:
We cannot do both. The JSON-based fields have to remain exported for the unmarshalling, so if we inlined `ManagerOem` into `Manager`, all the fields would be visible there. If we leave the extra indirection as-is, we can unexport the `Oem` field, and the JSON fields would all be unexported.

I think we should do it like this:

```
type Manager struct {
	redfish.Manager
	supermicro supermicro
}

type supermicro struct {
	ODataType string `json:"@odata.type"`
	NTP       common.Link
	Syslog    common.Link
	IKVM      common.Link
}

func FromManager(manager *redfish.Manager) (Manager, error) {
	type oemFields struct {
		Supermicro supermicro `json:"Supermicro"`
	}
	var oem oemFields
	err := json.Unmarshal(manager.Oem, &oem)
	if err != nil {
		return Manager{}, fmt.Errorf("can't unmarshal OEM Manager: %v", err)
	}

	return Manager{
		Manager:    *manager,
		supermicro: oem.Supermicro,
	}, nil
}
```

Reply:
@stmcginnis what do you think?

Reply:
Close, but I think this would be best:

```go
type Manager struct {
	redfish.Manager
	fanMode         string
	iKVM            string
	ipAccessControl string
	kcsInterface    string
	licenseManager  string
	mouseMode       string
	ntp             string
	radius          string
	smcrakp         string
	snmp            string
	smtp            string
	snooping        string
	sysLockdown     string
	syslog          string
}

type managerOem struct {
	Supermicro struct {
		ODataType       string `json:"@odata.type"`
		FanMode         common.Link
		IKVM            common.Link
		IPAccessControl common.Link
		KCSInterface    common.Link
		LicenseManager  common.Link
		MouseMode       common.Link
		NTP             common.Link
		RADIUS          common.Link
		SMCRAKP         common.Link
		SNMP            common.Link
		SMTP            common.Link
		Snooping        common.Link
		SysLockdown     common.Link
		Syslog          common.Link
	} `json:"Supermicro"`
}

func FromManager(manager *redfish.Manager) (Manager, error) {
	var oem managerOem
	err := json.Unmarshal(manager.Oem, &oem)
	if err != nil {
		return Manager{}, fmt.Errorf("can't unmarshal OEM Manager: %v", err)
	}

	return Manager{
		Manager:         *manager,
		fanMode:         oem.Supermicro.FanMode.String(),
		iKVM:            oem.Supermicro.IKVM.String(),
		ipAccessControl: oem.Supermicro.IPAccessControl.String(),
		kcsInterface:    oem.Supermicro.KCSInterface.String(),
		licenseManager:  oem.Supermicro.LicenseManager.String(),
		mouseMode:       oem.Supermicro.MouseMode.String(),
		ntp:             oem.Supermicro.NTP.String(),
		radius:          oem.Supermicro.RADIUS.String(),
		smcrakp:         oem.Supermicro.SMCRAKP.String(),
		snmp:            oem.Supermicro.SNMP.String(),
		smtp:            oem.Supermicro.SMTP.String(),
		snooping:        oem.Supermicro.Snooping.String(),
		sysLockdown:     oem.Supermicro.SysLockdown.String(),
		syslog:          oem.Supermicro.Syslog.String(),
	}, nil
}

func (manager *Manager) NTP() (*NTP, error) {
	return GetNTP(manager.GetClient(), manager.ntp)
}

func (manager *Manager) Syslog() (*Syslog, error) {
	return GetSyslog(manager.GetClient(), manager.syslog)
}

func (manager *Manager) IKVM() (*IKVM, error) {
	return GetIKVM(manager.GetClient(), manager.iKVM)
}
```

Reply:
I can do it like that, but it feels very repetitive to me. Why is the type of, for example Manager.fanMode important, when it's not exported anyway?

Reply:
You can omit the extra ones that you are not adding accessor methods for. I just put them in for completeness.

This is more repetitive for the implementation side so it's cleaner for the library consumer.

Reply:
In other words, we implement it once here so others never need to think about it.